hardware/DomoticzHardware.h

```diff
@@ -28,7 +28,7 @@ class CDomoticzHardwareBase : public StoppableTask
 
 	int m_HwdID = { 0 }; //must be uniquely assigned
 	bool m_bSkipReceiveCheck = { false };
-	unsigned long m_DataTimeout = { 0 };
+	int m_DataTimeout = { 0 };
```

Comment:
I would like to propose a declaration here that classes (and mainworker) can use instead of -1 ... because that does not say a lot.

#define TIMEOUT_IGNORE -1

so classes can use 

`m_DataTimeout = TIMEOUT_IGNORE;`

..

`if (pHardware->m_DataTimeout != TIMEOUT_IGNORE)`

I still don't get it why you need -1 ... if it is set to 0 (by default), there is no check on timeout
That should be sufficient ?

Hardware that does not receive anything, or not much (like only switch commands once in a while) should set 'm_bSkipReceiveCheck', so the datatimeout is ignored.
Actually this whole boolean could also be removed completely in the code, as users should not be allowed to specify a datatimeout, or should not do so
(Maybe we should call it 'receive timeout' in the user interface to clear things up)